cpp/daal/src/externals/service_stat_ref.h

```diff
@@ -174,6 +175,56 @@ struct RefStatistics<double, cpu>
                    __int64 method)
     {
         int errcode = 0;
+        daal::internal::ref::OpenBlas<double, cpu> blasInst;
+        double accWtOld = *nPreviousObservations;
+        double accWt    = *nPreviousObservations + nVectors;
+        DAAL_INT one    = 1;
+        char transa     = 'N';
+        char transb     = 'N';
+        double beta     = 0.0;
+        double alpha;
+        if (accWtOld != 0)
+        {
+            double * sumOld = daal::services::internal::service_malloc<double, cpu>(nFeatures, sizeof(double));
+            DAAL_CHECK_MALLOC(sumOld);
+            for (DAAL_INT i = 0; i < nFeatures; ++i)
+            {
+                sumOld[i] = sum[i];
+            }
+            // S_old S_old^t/accWtOld
+            alpha = 1.0 / accWtOld;
```

Comment:
Any checks for overflow?

Reply:
Does onedal have some macros to check floating point overflow?